### ISSUE TITLE
Swagger-client defaults the Accept header to one found in produces, which it can't parse

### ISSUE THREAD

**Issue by ponelat** [swagger-api/swagger-js#403]
I'm running the boilerplate code found in the README, and I noticed than when I fetch a resource, my response has an empty object. I noticed that the Content Type header to set to application/xml - which I'm guessing, our client isn't fond of parsing (why would it be ;).

This must have something to do with the newly merged [global consumes/produces](https://github.com/swagger-api/swagger-js/commit/e1578ed241965de07268951417f982ec333a2219)
What's happening, afaik - is that we're defaulting to the *first in the array* of **produces**, which is arbitrary and determined by the order in which the they're found in the spec.

This just so happens to be application/xml, in the petstore.json spec. Which is causing the response to be empty :disappointed: 

So what I propose is that..
We should only send an **Accepts header** to something the *client* can parse. Not one found in the produces array.

**Comment by fehguy** [comment]
We should really be overriding the global accept header with whatever is defined by the operation.  Is that not the case now?

**Comment by ponelat** [comment]
#### tl;dr
Here is an app, and the output.
https://gist.github.com/ponelat/534088a33ab5656cea7d

-----------

global produces/consumes works like we expect.
Global is overwritten by operation specific, and then defaults to application/json.

In the particular case of the petstore spec, under /pet/{petId} we have a produces array, with application/xml being the first.
https://github.com/ponelat/swagger-notes/blob/master/specs/petstore.json#L254-L257

Now according to our logic this is the mime type we want to set as headers['Accept'] when executing our operation - getPetById().

Only we don't want application/xml because our operation can't process that - *I beleive*

Here is the root of the issue.
https://github.com/ponelat/swagger-js/blob/develop_2.0/lib/client.js#L366
We're not overriding the default, and the default doesn't work :(
Am I making sense?

**Comment by ponelat** [comment]
I've dug deeper, and this is minor.
What I'll suggest (and perhaps put up an issue) is for the default consume/produces to be `application/json` **if** consumes/produces has one and obviously respecting the chain of defaults.

Currently we default to the first index when we default to an array of consumes/produces.
somethings like this...
```javascript
// Not production code, names are for clarity
var produces = arg.produces || operation.produces[0] || global.produces[0];
```

Instead of defaulting to `produces[0]`, I'd like to default to application/json if it exists.

I'll close this. I don't know how to get xml data, but that's my issue :D

**Comment by wing328** [comment]
I agree with your suggestion of defaulting to application/json if it exists. Other client SDKs should probably follow similar logic.
